R/validate_factory.R

```diff
@@ -1,143 +1,134 @@
-
 #' Inspect and validate the content of a factory
 #'
-#' This function can be used to inspect the content of a factory and make sure
-#' it looks fine. This includes various sanity checks listed in details. The
-#' function returns a list of error and warning messages.
+#' `validate_factory()` can be used to inspect the content of a factory and make
+#' everything looks fine. This includes various sanity checks listed in details
+#' that will error if a problem is found.
+#' 
+#' @inheritParams compile_reports
+#' @param allow_duplicates A logical indicating duplicate filenames are
+#'   allowed across folders.
 #'
 #' @details
-#' Checks ran on the factory include
-#' (the result of a failure is indicated in brackets):
-#'
-#' \itemize{
-#'
-#'  \item the directory exists [error]
-#'
-#'  \item all mandatory files exist: .here, .gitignore [error]
-#'
-#'  \item all mandatory folders exist: report_sources/ [error]
-#'
-#'  \item all .Rmd reports have unique names once outside their folders (to
-#'  avoid conflicts in outputs) [warning]
-#'
-#' }
-#'
+#' Checks run on the factory include:
+#'   * the factory directory exists;
+#'   * the factory_config file exist;
+#'   * all mandatory folders exist - by default these are 'report_sources/'
+#'     and 'outputs/';
+#'   * (optionally) all .Rmd reports have unique names once outside their
+#'     folders.
+#' 
+#' @return A list with 4 entries:
+#'   * root - the root folder path of the factory;
+#'   * factory_name - the name of the report factory;
+#'   * report_sources - the name of the report_sources folder; and
+#'   * outputs - the name of the outputs folder.
 #'
 #' @export
-#'
-#' @author Thibaut Jombart \email{thibautjombart@@gmail.com}
-#'
-#' @inheritParams compile_report
-#'
-#' @param warnings A logical indicating if warnings should be issued;
-#' defaults to \code{TRUE}.
-#'
-#' @param errors A logical indicating if errors should be issued;
-#' defaults to \code{TRUE}.
-#'
-#'
-
-validate_factory <- function(factory = getwd(),
-                             warnings = TRUE,
-                             errors = TRUE) {
-
-  out <- list(warnings = character(0),
-              errors = character(0))
-
-  ## check that the directory exists
-
-  if (!dir.exists(factory)) {
-      msg <- sprintf("the directory '%s' does not exist", factory)
-      stop(msg)
+validate_factory <- function(factory = ".", allow_duplicates = TRUE) {
+
+  # this finds the folder containing factory_config or errors if not possible
+  root <- factory_root(factory)
+
+  # load configuration
+  config <- as.data.frame(read.dcf(fs::path(root, "factory_config")))
+
```

Comment:
Petty comment again, but would it make sense to externalise this? Trying to think if we'll repeatedly need to check the config, or if we'll want to show it to the user. Note, we can always do this later.

Reply:
Let's think about this in the follow-up release.